templates/pkg/resource/resource.go.tpl

```diff
@@ -67,15 +72,38 @@ func (r *resource) SetObjectMeta(meta metav1.ObjectMeta) {
 }
 
 // SetIdentifiers sets the Spec or Status field that is referenced as the unique
-// resource identifier
+// resource identifier and any additional spec fields that may be required for
+// describing the resource.
 func (r *resource) SetIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
 {{- if $idField := .CRD.SpecIdentifierField }}
 	if identifier.NameOrID == nil {
 		return ackerrors.MissingNameIdentifier
 	}
 	r.ko.Spec.{{ $idField }} = identifier.NameOrID
 {{- else }}
+	if r.ko.Status.ACKResourceMetadata == nil {
+		r.ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
 	r.ko.Status.ACKResourceMetadata.ARN = identifier.ARN
 {{- end }}
+
+	if len(identifier.AdditionalKeys) == 0 {
+		return nil
+	}
+
+	specRef := reflect.Indirect(reflect.ValueOf(&r.ko.Spec))
+	specType := specRef.Type()
+
+	// Iterate over spec fields and associate corresponding json tags
+	for i := 0; i < specRef.NumField(); i++ {
+		// Get only the first field in the json tag (the field name)
+		jsonTag := strings.Split(specType.Field(i).Tag.Get("json"), ",")[0]
```

Comment:
I'm OK with using the `reflect` package (even though it's incredibly ugly to work with, IMHO) here.

In the future, we may want to take an approach similar to how we handle TerminalConditions and simply generate a `switch` case lookup here...